### ISSUE TITLE
Results from uirouter's resolve and getting the data manually are similar, but not the same.

### ISSUE THREAD

**Issue by jondthompson** [angular-ui/ui-router#2308]
The following two code snippets should result in the same data, but one is loaded prior to the state change and the other is loaded asynchronously. They're not.  ng-repeat works with both of them, but I switched to [angular-utils-pagination](https://github.com/michaelbromley/angularUtils/tree/master/src/directives/pagination#installation), and the latter code works, but the former code does not.

However, I don't see a difference in the arrays that are being generated, and they work with ng-repeat.
I'm going to crosspost this issue to angular-utils-pagination and ui-router as well, as it's not handling the difference properly, as I'm not certain whether [uirouter](https://github.com/angular-ui/ui-router) or [restangular](https://github.com/mgonto/restangular/) is in error for giving the wrong information in the first place.

#### doesn't work with angular-utils-pagination
```javascript
.state( 'users', {
    url:'/users',
    controller: 'UsersCtrl',
    templateUrl: 'users/Users.tpl.html',
    data:{ pageTitle: 'Users' },
    authenticate: true,
    resolve: {
      users: ['Users', function(Users){
        return Users.getList({embed:'roles,user_certifications'});
      }]
    }
  })
.controller( 'UsersCtrl', function UsersController( $scope, users ) {
  $scope.users = users;
})
.factory('Users', function(Restangular){
  return Restangular.service('users');
})
;
```

#### works with angular-utils-pagination
```javascript
.state( 'users', {
    url:'/users',
    controller: 'UsersCtrl',
    templateUrl: 'users/Users.tpl.html',
    data:{ pageTitle: 'Users' },
    authenticate: true
  })
.controller( 'UsersCtrl', function UsersController( $scope, Users ) {
  Users.getList().then(function(result){
   $scope.users = result;
   });
})
.factory('Users', function(Restangular){
  return Restangular.service('users');
})
;
```

**Comment by eddiemonge** [comment]
Is this making any difference `{embed:'roles,user_certifications'}` ?

These are also not returning the same things. One is returning a data set and the other is a promise. Try doing a `console.log(users)` right before (or after) the $scope assignment in the first one and `console.log(result)` in the second. You should see the difference.

Also, you should post the issue number to the other project here and this one there so it can be referenced. Closing as not a ui-router bug but an implementation issue.

**Comment by jondthompson** [comment]
The embed isn't related to the issue, and is a typo in my description. I've removed it from the original comment.

The first code example is resolving the promise before the controller is called, therefore a data set. The second is waiting for the data to be retrieved from the promise, and setting the resulting data set to the variable.

Once set, $scope.users should be the same in each example.

I just tested your hypothesis about the differing console.log() commands. I don't see any difference in console.log, yet it is completely broken in angular-utils-pagination. I'm guessing that it's an issue with restangular being slightly off when it returns the data in the first instance in a way that the pagination directive doesn't like. 

You're right though, uirouter is the least likely to be causing this problem.

Here's the other two places that I've crossposted this issue...
https://github.com/michaelbromley/angularUtils/issues/246
https://github.com/mgonto/restangular/issues/1252
